pkg/ddl/resource_group.go

```diff
@@ -258,18 +265,23 @@ func SetDirectResourceGroupRUSecondOption(resourceGroupSettings *model.ResourceG
 
 // SetDirectResourceGroupRunawayOption tries to set runaway part of the ResourceGroupSettings.
 func SetDirectResourceGroupRunawayOption(resourceGroupSettings *model.ResourceGroupSettings, opt *ast.ResourceGroupRunawayOption) error {
-	if resourceGroupSettings.Runaway == nil {
-		resourceGroupSettings.Runaway = &model.ResourceGroupRunawaySettings{}
-	}
 	settings := resourceGroupSettings.Runaway
 	switch opt.Tp {
 	case pmodel.RunawayRule:
 		// because execute time won't be too long, we use `time` pkg which does not support to parse unit 'd'.
-		dur, err := time.ParseDuration(opt.RuleOption.ExecElapsed)
-		if err != nil {
-			return err
+		if opt.RuleOption.ExecElapsed != "" {
+			dur, err := time.ParseDuration(opt.RuleOption.ExecElapsed)
+			if err != nil {
+				return err
+			}
+			settings.ExecElapsedTimeMs = uint64(dur.Milliseconds())
+		}
+		if opt.RuleOption.ProcessedKeys != 0 {
+			settings.ProcessedKeys = opt.RuleOption.ProcessedKeys
+		}
+		if opt.RuleOption.RequestUnit != 0 {
+			settings.RequestUnit = opt.RuleOption.RequestUnit
```

Comment:
```suggestion
		settings.ProcessedKeys = opt.RuleOption.ProcessedKeys
		settings.RequestUnit = opt.RuleOption.RequestUnit
```
Seems no need to check the if condition

Reply:
Since ProcessedKeys/RequestUnit/ExecElapsedTimeMs belongs to `RunawayOptionList` and `RunawayOptionList ` is a for loop, each Option has only one setting. Hence, we need to check here to avoid the last setting being useless.

Reply:
Ok. Then I suggest change `ResourceGroupRunawayRuleOption` to something like follows:
```
type ResourceGroupRunawayRuleOption {
    OpType RuleOptionType
    Value uint64
}
```

Reply:
How about just modifying `if` to `switch type` here?

Reply:
Another thought, as we only support add/update `RunawayRule` as a whole, why not use a unique `ResourceGroupRunawaySettings` in parser instead of the option list, I wonder what the benefit that this list bring us

Reply:
- `ExecElapsedTimeMs`, `ProcessedKeys`, `RequestUnit`, `ACTION` and `WATCH` belong `ResourceGroupRunawayOptionList`
- `RU_PER_SEC`, `PRIORITY`, ... , `QUERY_LIMIT` belongs the same level which using `ResourceGroupOptionList`
- `ResourceGroupRunawayOptionList` belongs `ResourceGroupRunawayOptionList`

With all splitted as `,` characters, it is difficult for the parser to determine the next character whether `ResourceGroupOptionList` or `ResourceGroupRunawayOptionList`.

Reply:
I think this is to support multiple out-of-order options?